aten/src/ATen/native/quantized/cpu/qadd.cpp

```diff
@@ -9,48 +9,62 @@
 namespace at {
 namespace native {
 namespace {
+
+inline Tensor requantize_into_int32(Tensor qa) {
+  if (qa.scalar_type() == kQInt32 && qa.q_scale() == 1.0 &&
+      qa.q_zero_point() == 0) {
+    return qa;
+  }
+  const auto a = qa.dequantize();
+  return at::quantize_linear(a, /*scale=*/1.0, /*zero_point=*/0, kQInt32);;
+}
+
+// Note: out is assumed to be the same size as self and other.
 template <bool ReLUFused = false>
-class QAddInt8 final : public c10::OperatorKernel {
- public:
-  Tensor operator()(
-      at::Tensor qa,
-      at::Tensor qb,
-      double scale,
-      int64_t zero_point) {
-    TORCH_CHECK(
-        qa.numel() == qb.numel(), "Add operands must be the same size!");
-    TORCH_CHECK(
-        qa.scalar_type() == qb.scalar_type(),
-        "Add operands should have same data type.");
-    auto a = qa.dequantize();
-    auto b = qb.dequantize();
-    auto c = at::empty_like(a);
-    auto iter = TensorIterator::binary_op(c, a, b);
+Tensor _add_out(Tensor& out, const Tensor& self, const Tensor& other) {
+  long zero_point = out.q_zero_point();
```

Comment:
s/long/int64_t/

long is platform-dependent and we use int64_t for everything in pytorch